rust/protocol/src/sealed_sender.rs

```diff
@@ -1194,14 +1491,18 @@ impl SealedSenderDecryptionResult {
     }
 }
 
+/// Decrypt a sealed sender message's sender information, then decrypt the underlying message.
+///
+/// Decrypt the sealed sender information from `ciphertext` using [sealed_sender_decrypt_to_usmc],
+/// drop the message if it was sent to the local user at `local_address`, then decrypt the
+/// underlying message according to [UnidentifiedSenderMessageContent::msg_type].
```

Comment:
I think this is at once more detail and less detail than a caller of this function would want. The self-send failure path is one of multiple kinds of errors that can happen, and the introduction of UnidentifiedSenderMessageContent doesn't really make sense without clicking through to `sealed_sender_decrypt_to_usmc`. (The one really important thing this method does that `sealed_sender_decrypt_to_usmc` does not is validate the embedded sender certificate; without that, you might have a forged certificate and therefore a forged sender.)

In another PR, it would probably be reasonable to take `local_address` as `Option<&ProtocolAddress>`, to skip the local address checking or reuse an existing object. That's too big a change for here, though. Stay on target!

Reply:
> (The one really important thing this method does that `sealed_sender_decrypt_to_usmc` does not is validate the embedded sender certificate; without that, you might have a forged certificate and therefore a forged sender.)

Thanks! I have removed this information entirely and focused the docstring purely on the meaning of the sender certificate validation (which is much more brief).

> In another PR, it would probably be reasonable to take `local_address` as `Option<&ProtocolAddress>`, to skip the local address checking or reuse an existing object. That's too big a change for here, though. Stay on target!

Thanks! I have reverted that change.